types/datasource.d.ts

```diff
@@ -76,28 +86,194 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
+
+  /**
+   * A hash-map of the different relation types.
+   */
+  static relationTypes: Record<string, string>;
+
+  constructor(name: string, settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
+
+  constructor(settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
 
   constructor(
     connectorModule: Connector,
-    settings?: Options,
+    settings?: ConnectorSettings,
     modelBuilder?: ModelBuilder,
   );
 
+  private setup(dsName: string, settings: ConnectorSettings): void;
+  private setup(settings: ConnectorSettings): void;
+
+  private _setupConnector();
+
+  /**
+   * Get the maximum number of event listeners
+   * 
+   * @remarks
+   * Defaults to {@link DataSource.DEFAULT_MAX_OFFLINE_REQUESTS} if not explicitly
+   * configured in {@link ConnectorSettings.maxOfflineRequests}.
+   */
+  getMaxOfflineRequests(): number;
+
+  // Reason for deprecation is not clear.
+  /**
+   * {@inheritDoc Connector.getTypes}
+   * @deprecated
+   */
+  getTypes(): string[];
+
+  /**
+   * Check if the datasource supports the specified types.
+   * @param types Type name(s) to check against
+   */
+  supportTypes(types: string | string[]): boolean;
+
+  /**
+   * Returns if the attached connector is relational.
+   * 
+   * @returns If the attached connector is relational; `undefined` if no
+   * connector is attached.
+   */
+  isRelational(): boolean | undefined;
+
+  freeze(): void;
+
+  /**
+   * Return the table name for the specified `modelName`.
+   * 
+   * @param modelName Target model name
+   * @returns The table name
+   */
+  tableName(modelName: string): string;
+
+  /**
+   * Retrieve the column name for the specified `modelName` and `propertyName`.
+   * 
+   * @param modelName Target model name
+   * @param propertyName Target property name
+   * @returns The column name
+   */
+  columnName(modelName: string, propertyName: string): string;
+
+  /**
+   * Retrieve the column names for the specified `modelName`.
+   * 
+   * @param modelName Target model name
+   * @returns Column names
+   */
+  columnNames(modelName: string): string;
+
+  /**
+   * Retrieve coulmn metadata for the specified `modelName` and `propertyName`.
+   * 
+   * @param modelName Target model name
+   * @param propertyName Target property name
+   * @returns Column metadata
+   */
+  columnMetadata(modelName: string, propertyName: string): ColumnMetadata;
+
+  /**
+   * Retrieve the ID property name for a model.
+   * 
+   * @param modelName Target model name
+   * @returns ID property name
+   */
+  idName(modelName: string): string;
+
+  /**
+   * Retrieve the ID property names sorted by their index.
+   * 
+   * @param modelName Target model name
+   * @returns Property names for IDs
+   */
+  idNames(modelName: string): string[];
+
+  /**
+   * Retrieve the ID property definition.
+   * 
+   * @param modelName Target model name
+   * @returns The ID property definition
+   */
+  idProperty(modelName: string): PropertyDefinition;
```

Comment:
There's also a `IdDefinition` in `model.d.ts`. Not sure if this function is supposed to return that instead.